homeassistant/components/electric_kiwi/sensor.py

```diff
@@ -0,0 +1,140 @@
+"""Support for Electric Kiwi sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import logging
+
+from electrickiwi_api.model import Hop
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.core import callback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import dt as dt_util
+
+from .const import ATTRIBUTION, DOMAIN
+from .coordinator import ElectricKiwiHOPDataCoordinator
+
+_LOGGER = logging.getLogger(DOMAIN)
+
+ATTR_EK_HOP_START = "hop_sensor_start"
+ATTR_EK_HOP_END = "hop_sensor_end"
+
+
+@dataclass
+class ElectricKiwiHOPRequiredKeysMixin:
+    """Mixin for required HOP keys."""
+
+    value_func: Callable[[Hop], datetime]
+
+
+@dataclass
+class ElectricKiwiHOPSensorEntityDescription(
+    SensorEntityDescription,
+    ElectricKiwiHOPRequiredKeysMixin,
+):
+    """Describes Electric Kiwi HOP sensor entity."""
+
+
+HOP_SENSOR_TYPE: tuple[ElectricKiwiHOPSensorEntityDescription, ...] = (
+    ElectricKiwiHOPSensorEntityDescription(
+        key=ATTR_EK_HOP_START,
+        name="Hour of free power start",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_func=lambda hop: datetime.combine(
+            datetime.today(), datetime.strptime(hop.start.start_time, "%I:%M %p").time()
+        ).astimezone(dt_util.DEFAULT_TIME_ZONE),
+    ),
+    ElectricKiwiHOPSensorEntityDescription(
+        key=ATTR_EK_HOP_END,
+        name="Hour of free power end",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_func=lambda hop: datetime.combine(
+            datetime.today(),
+            datetime.strptime(hop.end.end_time, "%I:%M %p").time(),
+        ).astimezone(dt_util.DEFAULT_TIME_ZONE),
+    ),
+)
+
+
+class ElectricKiwiHOPEntity(CoordinatorEntity, SensorEntity):
+    """Entity object for Electric Kiwi sensor."""
+
+    entity_description: ElectricKiwiHOPSensorEntityDescription
+
+    def __init__(
+        self,
+        hop_coordinator: ElectricKiwiHOPDataCoordinator,
+        description: ElectricKiwiHOPSensorEntityDescription,
+        customer_number: int,
+        connection_id: int,
+    ) -> None:
+        """Entity object for Electric Kiwi sensor."""
+        super().__init__(hop_coordinator)
+        self._hop_coordinator: ElectricKiwiHOPDataCoordinator = hop_coordinator
```

Comment:
Why set this again? The `CoordinatorEntity` already has the `coordinator` property available?